collector/app/core/models/transaction.py

```diff
@@ -0,0 +1,36 @@
+"""Module that includes functionality to work with transaction data."""
+
+import logging
+
+from asyncpg import exceptions
+
+from core.database.queries import INSERT_TRANSACTION
+
+
+LOG = logging.getLogger(__name__)
+
+
+class Transaction:
+    """Model that provides methods to work with transaction data."""
+
+    def __init__(self, postgres=None, redis=None):
+        """Initialize transaction instance with required clients."""
+        self.postgres = postgres
+        self.redis = redis
+
+    async def _get_mcc(self, code):
+        """Check if MCC exist in database, else return default value -1."""
+        mcc = await self.redis.get("mcc", deserialize=True, default=[])
+        if code not in mcc:
+            return -1
+
+        return code
+
+    async def save_transaction(self, user_id, transaction):
+        """Insert transaction element to postgres."""
+        transaction["mcc"] = await self._get_mcc(transaction["mcc"])
+        query = INSERT_TRANSACTION.format(user_id=user_id, **transaction)
```

Comment:
That is a bad approach. It will be vulnerable to SQL injections. We should use an approach like this:

ADD_CALLS_RECORD = """
INSERT INTO calls (calling, called, start, duration, rounded, price, cost)
VALUES ($1, $2, $3, $4, $5, $6, $7)
"""
await con.execute(
                ADD_CALLS_RECORD,
                call_object.calling,
                call_object.called,
                call_object.start.replace(tzinfo=None),
                call_object.duration,
                call_object.rounded,
                float(call_object.price),
                float(call_object.cost)
            )

Reply:
Good catch. Going to investigating this part.